src/plugins/event-plugins/JsErrorPlugin.ts

```diff
@@ -62,10 +65,13 @@ export class JsErrorPlugin implements Plugin {
     }
 
     private eventHandler = (errorEvent: ErrorEvent) => {
-        this.recordEvent(
-            JS_ERROR_EVENT_TYPE,
-            errorEventToJsErrorEvent(errorEvent, this.config.stackTraceLength)
+        const rumEvent = errorEventToJsErrorEvent(
+            errorEvent,
+            this.config.stackTraceLength
```

Comment:
question: Assuming we use a function filter, rather than a regex filter, what should the input to the `fitler` function be?

Specifically, do we want the `filter` parameter to be (1) the input to `eventHandler`, `promiseRejectEventHandler`, or `record`, or (2) the output of `errorEventToJsErrorEvent`?

Having the raw data (from eventHandler, promiseRejectEventHandler and record) would provide the most flexibility, although the `filter` function would potentially need to handle the most types.

I am in favor of forwarding the original error, to give devs the most flexibility to implement filters. We could mitigate some of the complexity by providing a sample implementation in the documentation.

Reply:
Hi, I wasn't expecting so many details in a code review. Great job!!!  To your question, I have some input: `forwarding original error event` would just work. Some non-standard implementation should be considered in a monitoring product because this is unpredictable if one put some extra data into an `Error` object or programmer throw string\number literal for some unknown reasons. By providing raw data, let the user makes decisions how to deal with an error captured.

Reply:
Again, great points - I do agree that if we are letting the customers decide which errors to filter out, we should pass in the untouched `ErrorEvent` object.